tuning: recommend colocation (#1289)
rancher/rancher-docs: docs/reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md

Use RKE2 for large scale use cases.

> ### Keep all Upstream Cluster Nodes co-located

Reviewer: ```suggestion
### Keep All Upstream Cluster Nodes Co-located
```

> Although Kubernetes is designed to run nodes and control plane components in different "zones" to attain high availability, doing so can have negative consequences on performance.

Reviewer: ```suggestion
To provide high availability, Kubernetes is designed to run nodes and control components in different zones. However, if nodes and control plane components are located in different zones, network traffic may be slower. 
```

> In particular, traffic beteween Rancher components and the Kubernetes API is sensitive to network latency, and etcd performance will also be negatively affected between its nodes.

Reviewer: ```suggestion
Traffic between Rancher components and the Kubernetes API is especially sensitive to network latency, as is etcd traffic between nodes.
```

> Therefore run all upstream node clusters in the same location, and particularly make sure latency between etcd nodes and Rancher is as low as possible.

Reviewer: ```suggestion
To improve performance, run all upstream node clusters in the same location. In particular, make sure that latency between etcd nodes and Rancher is as low as possible.
```
